Foundation/NSJSONSerialization.swift

```diff
@@ -35,12 +35,48 @@ public struct NSJSONWritingOptions : OptionSetType {
 
 public class NSJSONSerialization : NSObject {
     
-    /* Returns YES if the given object can be converted to JSON data, NO otherwise.
-    
-    Other rules may apply. Calling this method or attempting a conversion are the definitive ways to tell if a given object can be converted to JSON data.
+    /* Determines whether the given object can be converted to JSON.
+       Other rules may apply. Calling this method or attempting a conversion are the definitive ways
+       to tell if a given object can be converted to JSON data.
+       - parameter obj: The object to test.
+       - returns: `true` if `obj` can be converted to JSON, otherwise `false`.
      */
     public class func isValidJSONObject(obj: AnyObject) -> Bool {
-        NSUnimplemented()
+        func isValidJSONObjectInternal(obj: AnyObject) -> Bool {
+            // object is String or NSNull
+            if obj is String || obj is NSNull {
+                return true
+            }
+
+            // object is NSNumber and is not NaN or infinity
+            if let number = obj as? NSNumber {
+                if number.doubleValue.isInfinite || number.doubleValue.isNaN
+                    || number.floatValue.isInfinite || number.floatValue.isNaN {
+                    return false
+                }
```

Comment:
This would would probably be better implemented using `guard` to guarantee the early exit